src/groovy/org/transmartproject/db/dataquery/highdim/HighDimensionDataTypeResourceImpl.groovy

```diff
@@ -119,6 +119,23 @@ class HighDimensionDataTypeResourceImpl implements HighDimensionDataTypeResource
     }
 
     @Override
+    List<AssayColumn> retrieveAssays(List<AssayConstraint> assayConstraints) {
+        // Each module should only return assays that match
+        // the marker types specified, in addition to the
+        // constraints given
+        assayConstraints << new MarkerTypeConstraint(
+                platformNames: module.platformMarkerTypes)
+
+        def assayQuery = new AssayQuery(assayConstraints)
+        List<AssayColumn> assays = assayQuery.retrieveAssays()
+        if (assays.empty) {
+            throw new EmptySetException(
+                    'No assay satisfies the provided criteria.')
+        }
+        assays
+    }
```

Comment:
Rather than copy pasting code, `retrieveData` should call this method, especially given that the whole point of the exercise was having a method that returned the same assays in the same order in both methods. One calling the other would ensure that.